pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -42,6 +42,7 @@
 import org.apache.pinot.common.request.context.RequestContextUtils;
 import org.apache.pinot.common.tier.TierFactory;
 import org.apache.pinot.common.utils.config.TagNameUtils;
+import org.apache.pinot.segment.local.aggregator.ValueAggregatorFactory;
```

Comment:
This is not the correct `ValueAggregatorFactory`. We should import `org.apache.pinot.core.segment.processing.aggregator.ValueAggregatorFactory` which is used for `RealtimeToOfflineTask`

Reply:
ah good catch, I forgot there was the second one, will rework it around that

Reply:
I've gone back to using a `Set`, but used a `Set` of `AggregationFunctionType`, and added a comment in the core `ValueAggregatorFactory` to remind people to update this `Set`

Reply:
We should be able to use `ValueAggregatorFactory.getValueAggregator()`. `dataType` can be read from the schema

Reply:
> We should be able to use `ValueAggregatorFactory.getValueAggregator()`. `dataType` can be read from the schema

that one is in `pinot-core`, which isn't available in `pinot-segment-local`, I could add the dependency but that seemed like a big change